src/sql/workbench/common/editor/query/queryEditorInput.ts

```diff
@@ -346,6 +321,26 @@ export abstract class QueryEditorInput extends EditorInput implements IConnectab
 			}
 		}
 		this._onDidChangeLabel.fire();
+
+		this.contextualizeEditorForCopilot();
+	}
+
+	private async contextualizeEditorForCopilot(): Promise<void> {
+		// Don't need to take any actions if contextualization is not enabled and can return
+		if (!this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').githubCopilotContextualizationEnabled) {
```

Comment:
I'd even suggest moving this all into the contextualization service (make a single function `sendContextualizationToCopilot` that will handle getting and sending it) - that way all the copilot related stuff is all in one place which should help simplify some of this logic.